src/mbedtls/mbedtls_config.h

```diff
@@ -0,0 +1,4249 @@
+/**
+ * \file mbedtls_config.h
```

Comment:
How was this `mbedtls_config.h` file produced? It has a lot commented out, so are those maybe our custom changes?

It could be helpful to separate the introduction of pristine mbedTLS files into a separate first commit, and then make our changes with a subsequent commit.

Reply:
> I considered deleting large sections of the files instead of commenting things out (for readability).

I suggest adding `#if 0` around those sections instead of commenting out or deleting individual lines.

Reply:
Well if I'm adding a commit with pristine files, I definitely prefer deleting the large portions of unused/unrelated options in `mbedtls_config.h`. Way easier to read/digest (in the remote case anyone would want to).

Reply:
On second thought I'll leave that file as-is (lots of lines individually commented out), simply because that's how that file is normally used. But now that you can see the changes from the pristine file.